include/seqan3/io/alignment_file/format_sam.hpp

```diff
@@ -135,36 +135,81 @@ struct format_sam
 
     //!\brief The format version string.
     static constexpr char format_version[4] = "1.6";
-};
 
-} // namespace seqan3
+protected:
+    //!\privatesection
+    // Note that some member variables and functions are protected, because format_bam inherits from format_sam.
 
-namespace seqan3::detail
-{
+    //!\brief A buffer used when parsing arithmetic values with std::from_chars.
+    std::array<char, 316> buffer{}; // Doubles can be up to 316 characters
 
-//!\brief The seqan3::alignment_file_input_format_REMOVEME specialisation that handles formatted SAM input.
-//!\ingroup alignment_file
-template <>
-class alignment_file_input_format_REMOVEME<format_sam>
-{
-public:
-    //!\brief Exposes the format tag that this class is specialised with
-    using format_tag = format_sam;
+    //!\brief A variable that tracks whether the content of header has been written or not.
+    bool written_header{false};
```

Comment:
There are a bunch of functions in here that are also not part of the API. Since for this class the protected area is part of API that should be clearly separated. I would make a shared base class for sam and bam.